filebeat/prospector/prospector.go

```diff
@@ -214,12 +223,20 @@ func (p *Prospector) updateState(event *input.Event) error {
 	event.Module = p.config.Module
 	event.Fileset = p.config.Fileset
 
-	ok := p.outlet.OnEvent(event)
-	if !ok {
-		logp.Info("Prospector outlet closed")
-		return errors.New("prospector outlet closed")
+	eventMap := event.ToMapStr()
+
+	//run the filters before sending to
+	eventMap = p.filters.Run(eventMap)
+	if eventMap != nil {
```

Comment:
eventMap would be empty in case of `drop_event`. But i did see that the file was traversed till the end:

```
2017/03/28 08:17:17.442705 log.go:114: INFO Reader was closed: /var/log/alf.log. Closing.
2017/03/28 08:17:17.443151 log.go:306: DBG  Stopping harvester for file: /var/log/alf.log
2017/03/28 08:17:17.443163 log.go:315: DBG  Closing file: /var/log/alf.log
2017/03/28 08:17:17.443169 log.go:183: DBG  Update state: /var/log/alf.log, offset: 0
2017/03/28 08:17:17.443130 log.go:183: DBG  Update state: /var/log/displaypolicyd.stdout.log, offset: 1155
2017/03/28 08:17:17.443030 log.go:315: DBG  Closing file: /var/log/jamf.log
2017/03/28 08:17:17.443198 log.go:183: DBG  Update state: /var/log/jamf.log, offset: 1190762
```

But since these events do goto registrar, the registry file wouldn't be updated. But this is a scenario where I configure my prospector with:
```
filters:
  drop_event:
```

In this case, no event would pass upto registrar.